### PR TITLE
feat: mqtt based remotecontroller v1

### PR DESCRIPTION
**Pull request by l-spiecker**
Last Saturday evening I had time to implement a working prototype of this small nice-to-have feature. 

A network based presentation synchronization could help in different situations:
- Remote Control (with an App on a mobile device)
- View-only Instances (running on a Raspberry Pi at the beamer, or on a tablet eg. for keyboarder)
- for LiveStream lyrics with different Settings (font, background,..)
- for central SongStatistics with multiple instances on different PCs

My implementation is only a quick-and-dirty way of doing it with MQTT. Some classes are copied from Java 1.7/Android compatible Snippets I made a few years ago. The Observer Pattern could be changed to a JavaFX or bean based implementation. It is working, but Errors may occur by misconfiguration or connection instabilities.

Be blessed. ;-)

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -29,5 +30,14 @@
 		</attributes>
 	</classpathentry>
 	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4"/>
+	<classpathentry kind="lib" path="C:/Users/lenar/.m2/repository/org/eclipse/paho/org.eclipse.paho.client.mqttv3/1.1.0/org.eclipse.paho.client.mqttv3-1.1.0.jar" sourcepath="C:/Users/lenar/.m2/repository/org/eclipse/paho/org.eclipse.paho.client.mqttv3/1.1.0/org.eclipse.paho.client.mqttv3-1.1.0-sources.jar">
```
Comment (mathisdt) [inline .classpath:33]: Please don't use fixed paths.

File: pom.xml [hunk 2]
```diff
@@ -146,6 +146,13 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+		
+		<!--  REMOTE -->
+		<dependency>
+		    <groupId>org.eclipse.paho</groupId>
+		    <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
+		    <version>1.1.0</version>
```
Comment (mathisdt) [inline pom.xml:154]: Could you use at least version 1.2.1 of this MQTT client library? There's a security flaw in this version, see Snyk analysis of this commit.

**Comment by mathisdt** [comment]
Very nice! I was thinking of building a REST API for some time, but MQTT seems to be a better way to synchronize multiple devices. :)

BTW: I added some features lately (#48 #49 #50 #46) which might conflict with your changes. Could you merge them before marking this PR as ready? That would be kind!

File: .classpath (hunk 1 above)
Comment (l-spiecker) [reply .classpath]: Oh, of course. Comitted too fast.

**Comment by l-spiecker** [comment]
Implementing a REST Api would be also nice for remote control and presentation. I often choose MQTT for the retained messages and to get properties synchronized quickly. I think it is less suitable for large projects and web applications. Propably a Java EE stack would than fit better.
Ok, I'm gonna merge your changes this weekend.

File: pom.xml (hunk 2 above)
Comment (l-spiecker) [reply pom.xml]: Thanks. I changed the version now. I resolve this.

**Comment by mathisdt** [comment]
Something seems not to be right with the update notification. 

Master branch correctly doesn't display any notification in the upper right corner:
status-master

With this PR, the default text "status" seems to stay in the label. Normally it is removed when the checking takes place:
status-pr51

**Comment by l-spiecker** [comment]
Oh, let me fix this tomorrow. I think the status must be unset if remote is disabled, could be set on other connection issues or for not remote specific messages.

**Comment by l-spiecker** [comment]
Changed to a small status icon with a tooltip text. Shows the most relevant states.
grafik

**Comment by mathisdt** [comment]
I'm afraid the text "status" is still there if starting the SDB without configuration for remote access whatsoever. The icon is clearly better when using it, but without remoting the text should vanish or even not be there in the first place.

**Comment by l-spiecker** [comment]
I'm a bit confused, because i've removed the initial "status" text and the setText access.

See my latest commit in line 2849:
https://github.com/mathisdt/sdb2/pull/51/commits/0906ab5d4c4a43ff985d05d8b8013fc3cc753183#diff-d50ec8c2f5944478bba78c97dd479b5bL2849

grafik

**Comment by mathisdt** [comment]
Sorry, I messed up with my git remotes. It's all fine and can be merged.

**Pull request merged**